intake_bluesky/in_memory.py

```diff
@@ -48,32 +48,32 @@ def __init__(self, handler_registry=None, query=None, **kwargs):
         self._uid_to_run_start_doc = {}
         super().__init__(**kwargs)
 
-    def upsert(self, gen_func, gen_args, gen_kwargs):
-        gen = gen_func(*gen_args, **gen_kwargs)
-        name, run_start_doc = next(gen)
+    def upsert(self, gen_func, start_doc, get_stop, filename, gen_args, gen_kwargs):
```

Comment:
I had in mind `stop_doc` rather than `get_stop` because it is simpler (one less call in the stack) and conceptually easier, symmetric with `start_doc`. If there is no 'stop' document available, `stop_doc=None` can be given. If a 'stop' document is subsequently added, `upsert` will be called again the next time the catalog is reloaded, this time with `stop_doc={...}`.